model/lifecycle.go

```diff
@@ -55,25 +55,79 @@ type VersionToRestart struct {
 // SetVersionActivation updates the "active" state of all builds and tasks associated with a
 // version to the given setting. It also updates the task cache for all builds affected.
 func SetVersionActivation(versionId string, active bool, caller string) error {
-	builds, err := build.Find(
-		build.ByVersion(versionId).WithFields(build.IdKey),
-	)
+	version, err := VersionFindOneId(versionId)
+	if err != nil {
+		return errors.Wrapf(err, "getting version '%s'", versionId)
+	}
+	if version == nil {
+		return errors.Wrapf(err, "version '%s' not found", versionId)
+	}
+	if err = version.SetActivated(active); err != nil {
+		return errors.Wrapf(err, "setting activated for version '%s'", versionId)
+	}
+	changed, err := setTaskActivationForVersion(versionId, active, caller)
 	if err != nil {
-		return errors.Wrapf(err, "getting builds for version '%s'", versionId)
+		return errors.Wrapf(err, "setting activation for tasks in version '%s'", versionId)
 	}
-	buildIDs := make([]string, 0, len(builds))
-	for _, build := range builds {
-		buildIDs = append(buildIDs, build.Id)
+	if active && changed && evergreen.IsFinishedVersionStatus(version.Status) {
```

Comment:
So given that we're calling updateBuildStatus anyway, I think it does also maybe make reasonable sense to call UpdateVersionStatus and UpdatePatchStatus like we do for updating all builds: https://github.com/evergreen-ci/evergreen/blob/0aa9ce4746a3a31eb1d6cb9280c49e90d39b3efc/model/task_lifecycle.go#L1159 (really, I wonder if we should just call this function rather than calling updateBuildStatus explicitly at 108).

Reply:
calls UpdateVersionAndPatchStatusForBuilds